pkg/monocular/search.go

```diff
@@ -0,0 +1,132 @@
+/*
+Copyright The Helm Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package monocular
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"path"
+	"time"
+
+	"helm.sh/helm/pkg/chart"
+)
+
+// The structs below represent the structure of the response from the monocular
+// search API.
+
+// SearchResult represents an individual chart result
+type SearchResult struct {
```

Comment:
Do these structs exist in monocular and could we use them if so?

Reply:
They do. But, the package imports `k8s.io/helm/pkg/proto/hapi/chart`. I'd like to avoid that package import from helm v2 to helm v3. Any ideas other than copying?

Reply:
Nope, that makes complete sense